File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    a =[1,2,3,4,5]
    b = [2,1,2,3,2,4,2,5]
    c = [3,3,1,1,2,2,4,4,5,5]
    count_a = 0
    count_b = 0
    count_c = 0
    
    for i in range(len(answers)):
        if a[i % len(a)] == answers[i]:
            count_a +=1
        if b[i % len(b)] == answers[i]:
            count_b +=1
        if c[i % len(c)] == answers[i]:
            count_c +=1
           
    if count_a > count_b and count_a > count_c:
        return [1]
    elif count_b > count_a and count_b > count_c:
        return [2]
    elif count_c > count_b and count_c > count_a:
        return [3]


    if count_a == count_b == count_c:
        return [1,2,3]
    elif count_a == count_b:
        return [1,2]
    elif count_b == count_c:
        return [2,3]
    elif count_a == count_c:
        return [1,3]

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> one = {1,2,3,4,5};
    vector<int> two = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> three = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    int cnt1 =0;
    int cnt2 =0;
    int cnt3 =0;
    

    for(int i=0; i<answers.size(); i++){
        if(answers[i] == one[i%5]){
            cnt1 +=1;
        } 
        if (answers[i] == two[i%8]){
            cout << i%8;
            cnt2 +=1;
        }
        if (answers[i] == three[i%10]){
            cnt3 +=1;
        }
    }

    
    int max_cnt = max({cnt1, cnt2, cnt3});
    if (cnt1 == max_cnt){
        answer.push_back(1);
    } 
    if (cnt2 == max_cnt){
        answer.push_back(2);
    } 
    if (cnt3 == max_cnt){
        answer.push_back(3);
    }
    
     sort(answer.begin(), answer.end());
    return answer;
}


// 1. 정답 계산 -> 3번 돌려서 answer길이만큼고 맞춘 숫자만큼 count
// 2. max구하고 max가 몇명이지 확인 -> set에 넣는다(자동 오름차순) -> 반환
// 3.